### PR TITLE
ui/map: cleanup DestinationWidget

### PR DESCRIPTION
**Pull request by deanlee**
The original `DestinationWidget` was designed for `clearLayout`. refactored it to simplify code and improve performance. (No more dynamic properties and stylesheet)
The ui looks the same after the refactoring.

### CODE REVIEW

File: selfdrive/ui/qt/maps/map_settings.cc [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include "selfdrive/ui/qt/widgets/scrollview.h"
 
 MapSettings::MapSettings(bool closeable, QWidget *parent) : QFrame(parent) {
-  setContentsMargins(0, 0, 0, 0);
+  setAttribute(Qt::WA_NoMousePropagation);
```
Comment (sshane) [inline selfdrive/ui/qt/maps/map_settings.cc:11]: you can open a new PR for this clean up

File: selfdrive/ui/qt/maps/map_settings.h [hunk 2]
```diff
@@ -69,31 +66,27 @@ class MapSettings : public QFrame {
 class DestinationWidget : public QPushButton {
   Q_OBJECT
 public:
-  explicit DestinationWidget(QWidget *parent = nullptr);
-  void set(const QJsonObject &location, bool current = false);
-  void unset(const QString &label, bool current = false);
+  enum Type { Home, Work, Favorite, Recent, Current };
```
Comment (incognitojam) [inline selfdrive/ui/qt/maps/map_settings.h:69]: is there any reason to make `Type` public when the `typeIcon` helper is private?

Comment (deanlee) [reply selfdrive/ui/qt/maps/map_settings.h:69]: `class MapSettings` need access `Type` to create different `DestinationWidget`:

current_widget = new DestinationWidget(DestinationWidget::Current, this)
home_widget = new DestinationWidget(DestinationWidget::Home, this)
work_widget = new DestinationWidget(DestinationWidget::Work, this)

Comment (incognitojam) [reply selfdrive/ui/qt/maps/map_settings.h:69]: oh!

**Comment by incognitojam** [comment]
The action border radius has changed slightly

**Before**
image

**After**
image

**Comment by deanlee** [comment]
omg...Sorry I didn't check enough.

**Pull request closed without merging**